### ISSUE TITLE
🐛 `--skip-errors` CLI option is ignored

### ISSUE THREAD

**Issue by Obstbeeren** [biomejs/biome#3859]
### Environment information

```block
CLI:
  Version:                      1.9.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v22.7.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/4.0.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

The `--skip-errors` CLI option does not seem to work. When running `biome check ./file.js --skip-errors` on a file with syntax errors, biome still throws an error. According to the [documentation](https://biomejs.dev/reference/cli/#biome-check) it should just ignore the file.

```js
// file.js

((

))
```

### Expected result

It should not throw an error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
